src/renderer/screens/manager/AppsList/index.js

```diff
@@ -54,9 +60,17 @@ const AppsList = ({ deviceInfo, result, exec, t }: Props) => {
   const filteredState = omit(state, "currentProgress");
   const progress = state.currentProgress;
   const isIncomplete = isIncompleteState(filteredState);
-  const distribution = distribute(filteredState);
 
   const { installQueue, uninstallQueue, currentError } = filteredState;
+  const jobInProgress = installQueue.length > 0 || uninstallQueue.length > 0;
+
+  const distribution = useMemo(() => {
+    const newState = installQueue.length
+      ? predictOptimisticState(reducer(filteredState, { type: "install", name: installQueue[0] }))
```

Comment:
I don't understand why we do this. can you explain and/or comment in the code?

Reply:
For the optimistic device storage bar as you suggested.
It allows to show the size after the installs are complete

Reply:
makes sense!